landlab/components/radiation/radiation.py

```diff
@@ -211,50 +285,113 @@ def update(self):
+        self._ea = 0.6108 * np.exp((17.27 * self._Tmin) / (237.7 + self._Tmin))
 
-        self._delta = 23.45 * np.radians(
-            np.cos(2 * np.pi / 365 * (172 - self._julian))
-        )  # Declination angle
+        # Slope of Saturation Vapor Pressure - ASCE-EWRI Task Committee Report,
+        # Jan-2005 - Eqn 5, (36)
+        self._delta = (4098.0 * self._es) / ((237.3 + self._Tavg) ** 2.0)
 
+        # Solar Declination Angle - ASCE-EWRI Task Committee Report,
+        # Jan-2005 - Eqn 24,(51)
+        self._sdecl = 0.409 * np.sin(((np.pi / 180.0) * self._julian) - 1.39)
+
+        # Inverse Relative Distance Factor - ASCE-EWRI Task Committee Report,
+        # Jan-2005 - Eqn 23,(50)
+        self._dr = 1 + (0.033 * np.cos(np.pi / 180.0 * self._julian))
+
+        """ The following section computes the ratio to flat surface radiation field """
+        ####################################################################
+
+        # Apply distance factor to solar constant
+        # Correction of solar constant using distance,
+        # Src: https://www.lehman.edu/academics/eggs/documents/IJOC-2001-Becker.pdf
+        self._Io *= self._dr
+
+        # Convert latitude to radians and store trig calculations
+        self._phi = np.radians(self._latitude)  # Latitude in Radians
+        self._sinLat = np.sin(self._phi)
+        self._cosLat = np.cos(self._phi)
+
+        # Get the hour angle using time of day
         self._tau = (self._t + 12.0) * np.pi / 12.0  # Hour angle
 
+        # Calculate solar altitude using declination angle, hour angle, and azimuth
         self._alpha = np.arcsin(
-            np.sin(self._delta) * np.sin(self._phi)
-            + np.cos(self._delta) * np.cos(self._phi) * np.cos(self._tau)
-        )  # Solar Altitude
+            np.sin(self._sdecl) * self._sinLat
+            + (np.cos(self._sdecl) * self._cosLat * np.cos(self._tau))
+        )  # Solar Altitude/Angle
 
         if self._alpha <= 0.25 * np.pi / 180.0:  # If altitude is -ve,
             self._alpha = 0.25 * np.pi / 180.0  # sun is beyond the horizon
 
-        # Counting for Albedo, Cloudiness and Atmospheric turbidity
-        self._Rgl = self._Io * np.exp(
-            (-1)
-            * self._n
-            * (0.128 - 0.054 * np.log10(1.0 / np.sin(self._alpha)))
-            * (1.0 / np.sin(self._alpha))
+        # For less constant calculation, keeping solar altitude trig in fixed variables
+        self._cosSA = np.cos(self._alpha)
+        self._sinSA = np.sin(self._alpha)
+
+        # Avoid div by zero
+        if self._cosSA <= 0.0001:
+            self._cosSA = 0.0001
+
+        # self._hAboveSea = 300
+        self._hAboveSea = self._max_elev
+        self._temp = self._Tavg  # in C
+
+        # To calculate ws - ASCE-EWRI Task Committee Report,
+        # Jan-2005 - Eqn 29,(61)
+        self._x = 1.0 - (((np.tan(self._phi)) ** 2.0) * (np.tan(self._sdecl) ** 2.0))
+        if self._x <= 0:
+            self._x = 0.00001
```

Comment:
The equations in the ASCE standardized reference specifically call for this 0.00001 adjustment for when X is below or equal to zero to avoid division by zero errors, and there is nothing mentioning this specific case being an error of any sort, so this code should be kept as is. This situation arises when the tan^2 of the latitude and the solar declination angles are reciprocals which would yield an X value of zero.